### PR TITLE
feat: Create transaction from UI events

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description
Creates transactions when a control sends an action to the application, like button clicks and switch toggles.

## :bulb: Motivation and Context

Part of #1784 

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

_#skip-changelog_

### CODE REVIEW

File: Sources/Sentry/Public/SentryOptions.h [hunk 1]
```diff
@@ -198,6 +198,12 @@ NS_SWIFT_NAME(Options)
  * Default value is <code>NO</code>
  */
 @property (nonatomic, assign) BOOL attachScreenshot;
+
+/**
+ * When enabled, the SDK start transactions for UI Events like Buttons clicks, Switch toggle,
+ * Scrollview interaction and others.
```
Comment (philipphofmann) [inline Sources/Sentry/Public/SentryOptions.h:204]: I think we should define all UI events and remove others.
```suggestion
 * When enabled, the SDK creates transactions for UI events like buttons clicks, switch toggles,
 * and scroll view interactions.
```

File: Tests/SentryTests/SentryOptionsTest.m [hunk 2]
```diff
@@ -631,6 +631,11 @@ - (void)testAttachScreenshot
     [self testBooleanField:@"attachScreenshot" defaultValue:NO];
 }
 
+- (void)testEnableUserInteractionTrackin
```
Comment (philipphofmann) [inline Tests/SentryTests/SentryOptionsTest.m:634]:
```suggestion
- (void)testEnableUserInteractionTracking
```

**Review by philipphofmann — APPROVED** [review]
Thanks

File: Sources/Sentry/Public/SentryOptions.h (hunk 1 above)
Comment (brustolin) [reply Sources/Sentry/Public/SentryOptions.h:204]: Can you elaborate? I don't think I get it.

File: Sources/Sentry/SentryPerformanceTracker.m [hunk 3]
```diff
@@ -45,17 +45,28 @@ - (SentrySpanId *)startSpanWithName:(NSString *)name operation:(NSString *)opera
...
-        newSpan =
-            [SentrySDK.currentHub startTransactionWithContext:context
-                                                  bindToScope:SentrySDK.currentHub.scope.span == nil
-                                              waitForChildren:YES
-                                        customSamplingContext:@{}];
+
+        [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> span) {
+            BOOL bindToScope = true;
+            if (span != nil) {
+                if ([span.context.operation isEqualToString:@"ui.action"]) {
```
Comment (brustolin) [inline Sources/Sentry/SentryPerformanceTracker.m:58]: Im not sure about comparing the operation here.
Maybe we could have some kind of span level to determined whether we can override the current one in the scope or not.
But possible a future refactoring.

**Comment by philipphofmann** [comment]
> Hey @philipphofmann. besides the integration tests that I don't know why they are falling, I think this PR is good to merge so you can finish the other PR.

Feel free to merge.

**Pull request merged**